tools-local/unix/acquire-tool.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+
+usage()
+{
+    echo "usage: $0 <repository-root> <tool-name> [...]"
+    echo "  repository-root: Path to repository root."
+    echo "  tool-name: Name of the tool to download."
+    echo ""
+    echo "Downloads the declared version of the specified tool from the corresponding URL specified in the .toolversions file."
+    echo "If download succeeds then, returns the path to the executable."
+    echo "Exit 1 if download fails."
+}
+
+if [ -z "$1" ]; then
+    echo "Argument passed as repository-root is empty. Please provide a non-empty string."
+    usage
+    exit 1
+fi
+
+if [ -z "$2" ]; then
+    echo "Argument passed as tool-name is empty. Please provide a non-empty string."
+    usage
+    exit 1
+fi
+
+repoRoot="$1"
+toolName="$2"
+scriptPath="$(cd "$(dirname "$0")"; pwd -P)"
+. "$scriptPath/tool-helper.sh"
+declaredVersion="$(get_tool_config_value "$repoRoot" "$toolName" "DeclaredVersion")"
+
+
+# Downloads the package corresponding to the tool, and extracts the package.
+download_extract()
+{
+    # Get the download URL
+    downloadUrl="$(get_tool_config_value "$repoRoot" "$toolName" "DownloadUrl")"
+
+    if [ $? -ne 0 ]; then
+        echo "$downloadUrl"
```

Comment:
The name of the variable is rather confusing IMO.  It is not the downloadUrl in all cases.  May calling it a getDownloadUrlResult would make that more clear.  IDK.